3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CurrencyConversionEnrichment.scala

```diff
@@ -0,0 +1,157 @@
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[CurrencyConversionEnrichment] = {
+    isParseable(config, schemaKey).flatMap( conf => {
+      (for {
+        apiKey        <- ScalazJson4sUtils.extract[String](config, "parameters", "apiKey")
+        baseCurrency  <- ScalazJson4sUtils.extract[String](config, "parameters", "baseCurrency")
+        rateAt        <- ScalazJson4sUtils.extract[String](config, "parameters", "rateAt")
+        enrich        =  CurrencyConversionEnrichment(apiKey, baseCurrency, rateAt)
+      } yield enrich).toValidationNel
+    })
+  }
+}
+
+/**
+* Case class to wrap everything we
+* can extract as converted Transaction
+* Amounts.
+*/
+case class TransactionAmounts(
+  total: String, 
+  tax: String, 
+  shipping: String)
+
+// Object and a case object with the same name
+
+case class CurrencyConversionEnrichment(
+  apiKey: String,
+  baseCurrency: String, 
+  rateAt: String) extends Enrichment {
+
+  val version = new DefaultArtifactVersion("0.1.0")
+  
+  // To convert string value to double
+
+  def toDouble(s: String) = try { 
+    Some(s.toDouble) 
+  } catch { 
+    case _ => None 
+  }
+  
+  // To provide Validation for org.joda.money.Money variable
+  def eitherToValidation(input:Either[OerResponseError, Money]): Validation[String, Double]= {
+    input match {
+      case Right(l) => (l.getAmount().doubleValue()).success[String]
+      case Left(l) => (s"Open Exchange Rates error, message: ${l.errorMessage}").failure
+    }
+  }
+   
+  // To retrieve value from Success/Failure and parse them as string 
+  def tryParse(s: Either[OerResponseError, Money]):String = {
+    eitherToValidation(s) match {
+      case Success(pass) => {
+        pass.toString()
+      }
+      case Failure(fail) => {
+        fail
+      }
+    }
+  }
+
+  /**
+  * Convert's currency for a given
+  * set of currency, using
+  * Scala-Forex.
+  *
+  * @param trCurrency The desired
+  *        currency for a given 
+  *        amount
+  * @param trAmounts Contains
+  *        total amount, tax, shipping
+  *        amount's
+  * @param tiCurrency Trasaction Item
+  *        Currency
+   * @param tiPrice Trasaction Item
+  *         Price
+  * @return the converted currency
+  *         in TransacrionAmounts
+  *         format and Ttansaction
+  *         Item Price 
+  */
+  def convertCurrencies(trCurrency: String, trAmounts: TransactionAmounts, tiCurrency: String, tiPrice: String): ValidationNel[String, (TransactionAmounts, String)] = {
+    
+    try{
+      val fx = Forex(ForexConfig( nowishCacheSize = 0, nowishSecs = 0, eodCacheSize= 0, baseCurrency = baseCurrency), OerClientConfig(apiKey, DeveloperAccount))  
+      val newCurrencyTr = fx.convert((toDouble(trAmounts.total)).get).to(trCurrency).nowish
+      val newCurrencyTi = fx.convert((toDouble(tiPrice)).get).to(tiCurrency).nowish
+      val newTrAmountsTax = fx.convert((toDouble(trAmounts.tax)).get).to(trCurrency).nowish
```

Comment:
trCurrency is the currency we are converting from, not to. **Re-read how scala-forex works.**